cns/fsnotify/fsnotify.go

```diff
@@ -0,0 +1,141 @@
+package fsnotify
+
+import (
+	"context"
+	"io/fs"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type cnsclient interface {
+	ReleaseIPs(ctx context.Context, ipconfig cns.IPConfigsRequest) error
+}
+
+type Watcher struct {
+	CnsClient cnsclient
+}
+
+func WatchFs(w *Watcher, path, directory string, logger *zap.Logger) error {
+	// Create new fs watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("Error creating watcher: ", zap.Error(err))
+	}
+	defer watcher.Close()
+
+	// Start listening for events.
+	logger.Info("Listening for events from watcher")
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				logger.Info("Watcher: ", zap.Any("event: ", event.Name))
```

Comment:
don't use `zap.Any` when you know the type of the thing you're logging, 
the log message should be something useful, not just metadata (that should be in its own field attached to the logger)
```suggestion
				logger.Info("received event", zap.String("event", event.Name))
```